### PR TITLE
Skip read failures

### PR DESCRIPTION
**Pull request by skearnes**

### CODE REVIEW

File: rdkit_utils/serial.py [hunk 1]
```diff
@@ -206,15 +207,27 @@ def _get_mols(self):
...
             raise NotImplementedError('Unrecognized molecule format ' +
                                       '"{}"'.format(self.mol_format))
 
+        # skip read errors
+        while True:
+            try:
+                mol = mols.next()
+            except StopIteration:
+                break
+            except Exception:
```
Comment (rbharath) [inline rdkit_utils/serial.py:226]: Should we print a warning here? Silently skipping errors seems like it might cause issues.

**Comment by rbharath** [comment]
LGTM. Feel free to merge once you've addressed the comment.

**Comment by skearnes** [comment]
@rbharath Often rdkit will generate a warning about the specific issue, but without identifying the molecule that failed (although I think the SDF reader usually tries to provide a line number). I added some more specific output for SMILES (it will print out the offending line). 

This PR is really just a fix for getting random `None` molecules in the MolReader generator.

**Comment by rbharath** [comment]
Sounds reasonable. Feel free to merge.

**Pull request merged**